config/application.rb

```diff
@@ -17,5 +17,12 @@ class Application < Rails::Application
     # Application configuration can go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded after loading
     # the framework and any gems in your application.
+
+    config.middleware.insert_before 0, Rack::Cors do
+      allow do
+        origins 'localhost:3001'
```

Comment:
The point I'm trying to make is that this won't always be localhost. Only in dev. Rails supports different configs for different environments, so the url part of `origins 'localhost:3001'` should be a variable set in `config/development.rb`, `config/production.rb` etc